transcom/mymove
# MB-11210: Affiliation/Branch type other for customer onboarding

## [Jira ticket](https://dp3.atlassian.net/browse/MB-11210) for this change

## Summary

This PR adds the other type to customer onboarding, for those who are not associated with a specific branch of the military (e.g. retirees and separatees). This was already an option in the office app so no database changes were needed.

On the GHC/office side I updated the branch references so that they share the swagger definition with the internal one. 

## Verification Steps for Reviewers

- Create a new move with Branch type Other
- The branch type selected should show up in the review and profile page 
- Office users should be able to view this move and see the branch type (in the header and in the allowances section)
- Office users should be able to update someones branch type to/from Other
- All tests (client, server, e2e) are running and passing

### Frontend


## Screenshots
![Create your profile](https://user-images.githubusercontent.com/4184019/157305810-0fcb3e7e-7ba0-4f9f-a176-dcdeb850c2d3.png)
